core/src/main/java/org/jdbi/v3/core/EnumConfig.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.jdbi.v3.core.config.JdbiConfig;
+
+/**
+ * Configuration for behavior related to {@link Enum}s.
+ */
+public class EnumConfig implements JdbiConfig<EnumConfig> {
+    private static final Set<Class<? extends Annotation>> ALLOWED_HANDLINGS = new HashSet<>(Arrays.asList(EnumByName.class, EnumByOrdinal.class));
```

Comment:
This restricts the added qualifying annotations, which removes the ability to bind your own handling.  Is there a specific reason to forbid custom handling?  The purpose of allowing the user to set their own choice was to allow e.g. vendor extensions if they need to bind it as a special type of object, and this forbids that..

Reply:
Maybe instead we should just encourage them to register their own handler as a higher priority instead? 🤷‍♀️